### DISCUSSION TITLE
Creating an offline fallback page

### DISCUSSION THREAD

**Discussion by instaproadrian** [serwist/serwist#159] [Q&A]
Hi! Apologies for what is probably another dumb question and thanks for all the work you're doing on this!

I'm trying to build an offline fallback page for my app. I found next-pwa first, and then found serwist because you're recommending migrating over there. I like what I see so far - it seems more flexible and customisable. However, I'm struggling to work out how to implement a fallback page. next-pwa handles it out of the box for me and it seems to work, but serwist doesn't seem to. I also don't see info in the docs about how to do it.

Looking through the discussions I found [this comment](https://github.com/serwist/serwist/discussions/136#discussioncomment-9253597) suggesting I should do something like

```
const serwist = new Serwist({
  // more config
  fallbacks: {
    entries: [
      {
        url: "/_offline",
        matcher({ request }) {
          return request.mode === "navigate";
        },
      },
    ],
  },
});
```

Is that it? Or do I need more in `next.config.js`?

Thanks 🙌

**Comment by DuCanhGH** [comment]
@instaproadrian you will also need to add "/~offline" to `additionalPrecacheEntries` :) For revision, you may use anything that represents the page and is deterministic enough. I'd consider using `git rev-parse HEAD`.

Welp, perhaps we should improve the docs on this option. Or better yet, make the DX... better. Not sure how that is supposed to be done though, heh.
